### ISSUE TITLE
Object explorer uses old object definitions

### ISSUE THREAD

**Issue by LoganMallory** [rstudio/rstudio#2718]
### System details

    RStudio Edition : Desktop
    RStudio Version : Version 1.1.383
    OS Version      : OS X El Capitan 10.11.6
    R Version       : R version 3.4.3 (2017-11-30)

### Steps to reproduce the problem
See the GIF below.

I can not find any clean way to reproduce the problem. It seems to occur more frequently with large lists (the one in the GIF is ~52mb).

### Describe the problem in detail
Creating a list of data frames, call it list1, then deleting (removing) that list and initializing list1 as list1 <- list(), will fill list1 with the data frames it used to have instead of initializing it as an empty list.

### Describe the behavior you expected
list1 <- list() #should initialize list1 as an empty list, not a list of data frames.

![rstudiolistbug_gif](https://user-images.githubusercontent.com/38388056/39443573-0397bab4-4c83-11e8-9be0-46bb5cb921eb.gif)

**Comment by jmcphers** [comment]
@kevinushey can you take a look at this one? I repro'ed with Ron's instructions. R is not doing anything wrong here; the problem is that the object explorer is using the cached `df_vec`. `View(df_vec)` should refresh the object explorer, but doesn't. (The object explorer's Refresh button doesn't do anything either, and should).

**Comment by kevinushey** [comment]
I wasn't able to reproduce with @ronblum's instructions, but IIUC these are the repro steps:

1. Create and view an object with e.g.

```
object <- list(alpha = 1, beta = 2)
View(object)
```

2. Create a new definition for the object e.g.

```
object <- list(gamma = 3, delta = 4)
```

3. Return to the opened object explorer, and try clicking the Refresh button.

The old definition for `object` is used rather than the one in the global environment.

**Comment by dfalty** [comment]
Verified fixed on version 1.2.1090.
